### PR TITLE
Fix: version contexts via URL wouldn't persist

### PR DESCRIPTION
**Pull request by attfarhan**
Version contexts in localStorage should be updated when a URL uses a different version context. It wasn't occurring previously.

edit: added commit to make `setVersionContext` accept undefined, so callers can pass it instead of an empty string.

### CODE REVIEW

File: web/src/search/results/SearchResults.tsx [hunk 1]
```diff
@@ -193,9 +193,8 @@ export class SearchResults extends React.Component<SearchResultsProps, SearchRes
                                             if (caseSensitive !== this.props.caseSensitive) {
                                                 this.props.setCaseSensitivity(caseSensitive)
                                             }
-                                            if (versionContext !== this.props.versionContext) {
-                                                this.props.setVersionContext(versionContext || '')
-                                            }
+
+                                            this.props.setVersionContext(versionContext || '')
```
Comment (felixfbecker) [inline web/src/search/results/SearchResults.tsx:197]: Instead of an empty string, should this allow `undefined`?

Comment (attfarhan) [reply web/src/search/results/SearchResults.tsx]: Yep, just made `setVersionContext` accept undefined 👍

File: web/src/Layout.tsx [hunk 2]
```diff
@@ -113,7 +113,7 @@ export interface LayoutProps
         versionContext: string | undefined,
         { extensionsController }: ExtensionsControllerProps<'services'>
     ) => Observable<GQL.ISearchResults | ErrorLike>
-    setVersionContext: (versionContext: string) => void
+    setVersionContext: (versionContext: string | undefined) => void
```
Comment (felixfbecker) [inline web/src/Layout.tsx:116]: Usually these are called `onVersionContextChange` (the component doesn't know what happens in response to that, like setting state)

Comment (attfarhan) [reply web/src/Layout.tsx:116]: Oops saw this after I hit merge. Will clean up separately.

**Pull request merged**